app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -583,73 +570,94 @@ fun MapScreen(
-                      OverviewViewportStateOptions.Builder()
-                          .geometry(mapViewportState.cameraState!!.center)
-                          .padding(EdgeInsets(100.0, 100.0, 100.0, 100.0))
-                          .build())
-                  mapViewModel.updateTrackingMode(false)
-
-                  mapViewportState.setCameraOptions {
-                    center(
-                        Point.fromLngLat(
-                            suggestion.longitude.toDouble(), suggestion.latitude.toDouble()))
-                  }
+            }
+      }
+}
 
-                  Log.e(
-                      "Selected Location",
-                      Point.fromLngLat(
-                              suggestion.longitude.toDouble(), suggestion.latitude.toDouble())
-                          .toString())
-                },
-                colors =
-                    CardColors(
-                        containerColor = invertColor(defaultOnColor()),
-                        contentColor = defaultOnColor(),
-                        disabledContainerColor = invertColor(defaultOnColor()),
-                        disabledContentColor = defaultOnColor()),
-                modifier = Modifier.fillMaxSize().testTag("suggestionCard${suggestion.city}")) {
-                  Text(
-                      text =
-                          suggestion.suggestionFormatDisplayName(
-                              MAX_SUGGESTION_DISPLAY_NAME_LENGTH_MAP, Address.Mode.MAP),
-                      Modifier.padding(5.dp),
-                      textAlign = TextAlign.Start)
-                }
+/** Composable function to display the drag handle for the suggestion menu. */
+@Composable
+fun SuggestionsMenuDragHandle() {
+  Row(
+      modifier =
+          Modifier.padding(vertical = 12.dp)
+              .width(32.dp)
+              .height(4.dp)
+              .background(
+                  color = MaterialTheme.colorScheme.primary.copy(alpha = 0.4f),
+                  shape = RoundedCornerShape(2.dp)),
+      horizontalArrangement = Arrangement.Center) {}
+}
 
-            androidx.compose.material.Divider(Modifier.fillMaxWidth(), color = Black)
-          }
-        }
-      }
+/**
+ * Function to handle the click on a suggestion. It will center the camera on the suggestion and
+ * update the tracking mode to false.
+ *
+ * @param suggestion The suggestion clicked.
+ * @param showSuggestions The state of the suggestion menu.
+ * @param mapViewportState The viewport state of the map.
+ * @param mapViewModel The ViewModel for managing map-related data and actions.
+ */
+private fun handleSuggestionClick(
+    suggestion: Address,
+    showSuggestions: MutableState<Boolean>,
+    mapViewportState: MapViewportState,
+    mapViewModel: MapViewModel
+) {
+  showSuggestions.value = false
+
+  mapViewportState.transitionToOverviewState(
+      OverviewViewportStateOptions.Builder()
+          .geometry(mapViewportState.cameraState!!.center)
```

Comment:
100% sure that the cameraState can't be null? (I know it was like this before just ensuring that your new implementation doesn't change things w.r.t that)

Reply:
I don't know. I just took the previous code and put it into a function for modularization purposes. Didn't try to understand what it does in details. Just big picture to add documentation

Reply:
This is something you should maybe discuss with the author of this code and their reviewers?